11-integrating-with-function-calling/translations/ja-jp/README.md

```diff
@@ -0,0 +1,450 @@
+  1.`properties` - 回答に使用する特定の値のリスト
+      1. `name` - フォーマットされた回答で使用するプロパティ名。例えば、`product`
+      1. `type` - プロパティのデータ型。例えば、`string`
+      1. `description` - 特定のプロパティの説明
+
+また、オプションのプロパティとして `required`もあります。これは関数呼び出しが必須かを指定するプロパティです。
+
+### ステップ3 - 関数呼び出しの実行
+
+関数を定義した後、次に Chat Completion API の呼び出しにその関数を含める必要があります。これを行うには、リクエストに `functions` を追加します。この場合、`functions=functions`とします。
+
+また、`function_call` を `auto` に設定するオプションもあります。これは、どの関数を呼び出すか、利用者からのメッセージに基づいて LLM に判断させ、自分で割り当てるのではなく、LLM に選択させます。
+
+以下に、`ChatCompletion.create` を呼び出すコードを示します。`functions=functions` と `function_call="auto"` を設定し、LLM に提供する関数をいつ呼び出すかを選択させる方法に注目してください。
+
+```python
+response = client.chat.completions.create(model=deployment,
+                                        messages=messages,
+                                        functions=functions,
+                                        function_call="auto")
+
+print(response.choices[0].message)
+```
+
+返される回答は下記のようになります。
+
+```json
+{
+  "role": "assistant",
+  "function_call": {
+    "name": "search_courses",
+    "arguments": "{\n  \"role\": \"学生\",\n  \"product\": \"Azure\",\n  \"level\": \"初心者\"\n}"
+  }
+}
+```
+
+ここでは、`search_courses` 関数がどのように呼び出され、どのような引数で呼び出されたかを、JSON 形式で返された応答の `arguments` プロパティ・リストから確認できます。
+
+結論として、LLM は Chat Completion 呼び出しの `messages` パラメータで提供した値からデータを抽出し、関数の引数に適合するデータを見つけることができました。以下は、`messages` 値の内容です。
+
+```python
+messages= [ {"role": "user", "content": "初心者の学生が Azure を学ぶための良い学習コースを見つけてください。"} ]
+```
+
+ご覧の通り、`学生`、`Azure`、`初心者`は `messages` から抽出され、関数への入力として設定されました。このように関数を使用することは、プロンプトから情報を抽出するだけでなく、LLM に構造を提供し、再利用可能な機能を持たせる事ができます。
+
+次に、この機能をアプリでどのように使用できるかを確認します。
+
+## アプリケーションへ関数呼び出しの統合
+
+LLM からフォーマットされた応答をテストした後、これをアプリケーションに統合できます。
+
+### フローの管理
+
+アプリケーションに統合するためには、下記の手順で行います。
+
+1. まず、Open AI サービスを呼び出し、メッセージを `response_message` という変数に格納します。
+
+   ```python
+   response_message = response.choices[0].message
+   ```
+
+1. 次に、Microsoft Learn API を呼び出してコースの一覧を取得する関数を定義します。
+
+   ```python
+   import requests
+
+   def search_courses(role, product, level):
+     url = "https://learn.microsoft.com/api/catalog/"
+     params = {
+        "role": role,
+        "product": product,
+        "level": level
+     }
+     response = requests.get(url, params=params)
+     modules = response.json()["modules"]
+     results = []
+     for module in modules[:5]:
+        title = module["title"]
+        url = module["url"]
+        results.append({"title": title, "url": url})
+     return str(results)
+   ```
+
+   `functions` 変数に記載した関数名にマップする、実際の Python 関数を作成する方法に注目してください。また、必要なデータを取得するために、実際に外部 API 呼び出しも行っています。ここでは、Microsoft Learn API に対してトレーニング モジュールを検索しています。
+
+いま、`functions` 変数とそれに対応する Python 関数を作成しましたが、実際の Python 関数を呼び出すために、これら 2 つをマッピングした内容を、LLM に指示するにはどうすればよいでしょうか？
+
+1. To see if we need to call a Python function, we need to look into the LLM response and see if `function_call` is part of it and call the pointed out function. Here's how you can make the mentioned check below:
```

Comment:
```suggestion
```

You've already translated this part into Japanese at [line 331](https://github.com/microsoft/generative-ai-for-beginners/pull/230/files#diff-653d8b1f89858cccd0d260e78b0f6051f3dec8ffc975554f7ef01e5d20791629R331) , so I think the original text here isn't necessary. 😄